colossalai/quantization/fp8.py

```diff
@@ -0,0 +1,105 @@
+import torch
+import torch.distributed as dist
+
+
+def cast_to_fp8(inp: torch.Tensor, scale=None, fp8_format="e4m3") -> (torch.Tensor, torch.Tensor):
+    r"""
+    casting torch Tensor into specified fp8 tensor with per-channel scaling or per-tensor scaling.
+    Args:
+        inp: input torch Tensor, should be in torch.FloatTensor, torch.HalfTensor, torch.BFloat16Tensor.
+        scale: scaling factor for fp8 casting. If it is None, then it is computed automatically. Per-channel scaling
+        is applied if input tensor is 2 dimension, otherwise, per-tensor scaling is applied.
+        fp8_format: e4m3 or e5m2
+
+    Returns:
+        Tuples: A tuple (fp8_tensor, scale)
+    """
+    if inp.dtype in [torch.FloatTensor, torch.HalfTensor, torch.BFloat16Tensor]:
+        return inp
+    fp8_type = torch.float8_e4m3fn if fp8_format == "e4m3" else torch.float8_e5m2
+
+    if inp.dim() == 2:
+        if scale is None:
+            per_channel_max = inp.abs().max(dim=-1).values
+            scale = per_channel_max
```

Comment:
To reduce accuracy loss in fp8 conversion. I think scale should be per_channel_max/fp8_max instead of per_channel_max